spec/index.html

```diff
@@ -10424,7 +10424,205 @@ <h3>Evaluation Semantics</h3>
+F(v) = v if v is not in PV</pre>
+                <p>
+                Define the Projection Expression Variable Remapping `ProjectMap(P, PV)`
+                </p>
+                <pre>ProjectMap(Project(A, PV)) = Project(A1, PV) 
+                                where A1 is the result of applying F
+                                to every variable mentioned in A.
+                </pre>
+                <p>
+                  The Projection Expression Variable Remapping yields an algrebra expression that
+                  evaluates to the same results as the Project argument. No variable of `ProjectMap(Project(A, PV))`
+                  that is not in `PV` is mentioned anywhere else in the algebra expression for the query.
+                </p>
+              </div>
+              <p>This process is applied throughout the graph pattern of <code>EXISTS</code>:</p>
+              <div class="defn">
+                <b>Definition: <span id="defn_varrename" name="defn_varrename">Variable Remapping</span></b>
+                <p>
+                  For any algebra expression `X`, define the Variable Remapping `PrjMap(X)` 
+                  of algebra expression `X`:
+                </p>
+                  <pre>PrjMap(X) = replace all project operations Project(P, PV) 
+                with ProjectMap(P, PV) for each projection in X.</pre>
+              </div>
+              <p>
+                The outcome of `PrjMap` is independent of the order of replacement
+                (e.g. bottom-up or top-down).
+                Replacements may happen several times, depending on recursive order
+                but each time a replacement is made, the variable not used anywhere else.
+              </p>
+
+              <div class="note">
+                <p>
+                  A variable inside a project expression that is not in the variables projected
+                  is not affected by the values insertion operation because it is renamed apart.
+                </p>
+                <p>
+                  This operation is as part of the <a href="#sparqlQuery">translation to the SPARQL
+                  algebra</a>.
+                </p>
+              </div>
+            </section>
+            <section>
+              <h4>Values Insertion</h4>
+              <p>
+                Alternative 1: rewrite the algebra during 
+                <a href="#sparqlQuery">translation to the SPARQL algebra</a>
+                to include a function that evaluates to the current row.
+              </p>
+
+              <div class="defn">
+                <div>
+                  <b>Definition: <span id="defn_correlate" name="defn_correlate">Values In Scope</span></b>
+                  <p>
+                    Define the function `BindingInScope()`. 
+                    Evaluation of `BindingInScope()` results in a table of one row,
+                    being the current binding of the enclosing filter.
```

Comment:
EXIST can be used not only within FILTER.

Reply:
Bind, select expression, order by…

Reply:
Added: https://github.com/afs/SPARQL-exists/tree/main/tests/exists-6-places